optimus/wdl_tasks/StarAlignFastqSingleEnd.wdl

```diff
@@ -0,0 +1,39 @@
+
+task StarAlignFastqSingleEnd {
+  File fastq_input
+  File gztar_star_reference
+
+  # estimate input requirements for star using UBams:
+  # multiply input size by 2.2 to account for output bam file + 20% overhead, add size of reference.
+  Int input_size = ceil(size(fastq_input, "G"))
+  Int reference_size = ceil(size(gztar_star_reference, "G"))
+  Int estimated_disk_required =  ceil(input_size * 2.2 + reference_size)
+
+  command {
+    # prepare reference
+    mkdir genome_reference && tar -xzf ${gztar_star_reference} -C genome_reference --strip-components 1
+    rm ${gztar_star_reference}
+
+    STAR \
+      --runMode alignReads \
+      --runThreadN 16 \
```

Comment:
If you replace 16 with `$(nproc)` you won't have to worry about remembering to change this if you change the instance type.

Reply:
Great idea. Will change.